src/modules/mount/main.py

```diff
@@ -294,6 +299,9 @@ def mount_partition(root_mount_point, partition, partitions, mount_options, moun
         # Mount the subvolumes
         swap_subvol = libcalamares.job.configuration.get("btrfsSwapSubvol", "/@swap")
         for s in btrfs_subvolumes:
+            # If the subvolume is "", don't try to mount it.
+            if not s["subvolume"]:
+                continue
             if s['subvolume'] == swap_subvol:
```

Comment:
Looking at it in total, in this part of the change all we need?

Reply:
I thought that still might be wanted, for any subvolumes where the subvolume is "" (which in practice should only be the / mountPoint, but may change at some point, I'm not sure) we probably don't want to try to mount the subvolume, right?

Reply:
That said if it causes confusion I'm happy to remove that bit. It's not strictly needed for my use case.

My use case being just supporting this in mount.conf:

btrfsSubvolumes:
    - mountPoint: /
      subvolume: ""

Reply:
When the subvolume is `""`, it means "The root of the partition is mounted on `/`".  I think that is the only thing this can ever mean because we only process the subvolume logic for the partition mounted at `/`.

However, we have already mounted the partition at `/` so I think we just need to "do nothing" for that subvolume.

Reply:
> That said if it causes confusion I'm happy to remove that bit. It's not strictly needed for my use case.

I think this part is all we need.  It is the other parts that aren't needed.

Reply:
What I was seeing in my testing here was that with the mount.conf I uploaded before without any changes to calamares sources the / mountPoint was mounted, then the subvolume logic would unmount / then rsync would try to rsync the rootfs to /tmp/calamares-foo which was just part of tmpfs, would fill up and fail. So something needs to drop out of this subvolume code and not do the umount either... right?

To fix that I added the dictionary comparison check at the top. I'll remove this bit in the subvolume mounting though since that's not needed for my use case. One moment.

Reply:
I pushed up just doing the config check and dropping out early if it matches. Though I haven't yet tested to verify that is how the above mentioned config ends up in dictionary form. I'll test that shortly.

Reply:
> When the subvolume is `""`, it means "The root of the partition is mounted on `/`". I think that is the only thing this can ever mean because we only process the subvolume logic for the partition mounted at `/`.
> 
> However, we have already mounted the partition at `/` so I think we just need to "do nothing" for that subvolume.

Do nothing but also not do the umount command.

Reply:
I see, I misread your comment as "Do we still need this with the other part"